### ISSUE TITLE
[a11y]: ContainedList [with expandable search] - search edit field incorrect focus behavior

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13941]
### Package

@carbon/react

### Browser
FF

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology
JAWS

### Description
The search field is triggered on down arrow navigation instead of user pressing enter on search button

### WCAG 2.1 Violation
3.2.1 On Focus

### Reproduction/example
Record `Number 29` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

In JAWS with FF, The search edit field is triggered when the user presses down arrow from the "search" button. This behavior confuses the blind screen reader user. Because the user may think that the search field is always present without requiring to activate the search button.

### User Impact
When an element automatically changes after receiving focus, it disorients users who have low vision or cognitive disabilities.

### Recommended fix
The search field is triggered on down arrow navigation instead of user pressing enter on search button

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by andreancardona** [comment]
Closed by https://github.com/carbon-design-system/carbon/pull/14076
